http-clients/apache-client/src/test/java/software/amazon/awssdk/http/apache/ApacheHttpClientWireMockTest.java

```diff
@@ -148,4 +158,52 @@ public void clear() {
 
         mockProxyServer.verify(2, RequestPatternBuilder.allRequests());
     }
+
+    @Test
+    public void overrideDnsResolver_withDnsMatchingResolver_successful() throws Exception {
+        overrideDnsResolver("magic.local.host");
+    }
+
+    @Test(expected = UnknownHostException.class)
+    public void overrideDnsResolver_withLocalhostIp_successful() throws Exception {
+        overrideDnsResolver("sad.local.host");
+    }
+
+    private void overrideDnsResolver(String hostName) throws IOException {
```

Comment:
`hostName` looks unused

Reply:
will take a look

Reply:
It's used for the URI a bit further down